lib/rubygems/commands/build_command.rb

```diff
@@ -9,6 +9,10 @@ def initialize
     add_option '--force', 'skip validation of the spec' do |value, options|
       options[:force] = true
     end
+
+    add_option '--[no-]allow-warnings', 'allow warnings when validating the spec' do |value, options|
```

Comment:
Here are some examples of how to test this option: https://github.com/rubygems/rubygems/blob/d3db595be39639fdc6b020e24fab4d0cf052b448/test/rubygems/test_gem_commands_check_command.rb#L29-L45